nnsvs/bin/prepare_features.py

```diff
@@ -105,44 +156,40 @@ def my_app(config : DictConfig) -> None:
     if config.timelag.enabled:
         logger.info("Timelag linguistic feature dim: {}".format(in_timelag[0].shape[1]))
         logger.info("Timelag feature dim: {}".format(out_timelag[0].shape[1]))
-        for idx in tqdm(range(len(in_timelag))):
-            x, y = in_timelag[idx], out_timelag[idx]
-            name = splitext(basename(in_timelag.collected_files[idx][0]))[0]
-            xpath = join(in_timelag_root, name + "-feats.npy")
-            ypath = join(out_timelag_root, name + "-feats.npy")
-            np.save(xpath, x, allow_pickle=False)
-            np.save(ypath, y, allow_pickle=False)
+        joblib.Parallel(n_jobs=-2)(
+            joblib.delayed(_prepare_timelag_feature)(
+                in_timelag_root, out_timelag_root, in_timelag, out_timelag, idx
+            )
+            for idx in tqdm(range(len(in_duration)))
+        )
```

Comment:
Could you replace `joblib.Parallel` to `concurrent.futures.ProcessPoolExecutor`? The later is more tqdm friendly. We cannot easily make tqdm work if we use joblib.Parallel.

You can refer to my another repository: https://github.com/r9y9/ttslearn/blob/23a4088571ac680f7e28b6ac59d3e2eb31b9511f/recipes/dnntts/preprocess_acoustic.py#L95-L110

Reply:
Also, it would be nice if we add `n_jobs` in config.yaml so that we can control it through the CLI (i.e., run.sh or prepare_features.py).